src/Chapter3.hs

```diff
@@ -910,6 +1021,22 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append (Gold x) (Gold y) = Gold (x + y)
+
+instance Append [a] where
+  append :: [a] -> [a] ->[a] 
+  --append a b = a ++ b -- eta reduce
+  append = (++) -- @TODO : Find how it infix operator style function are implmented
```

Comment:
That's very simple actually 🙂 
Functions in Haskell can be used in two forms: prefix and infix.

By default, named functions are used in prefix form:

```haskell
append list1 list2
```

and operators by default are used in infix form:

```haskell
x + y
```

However, you can use named functions in the infix form by taking function name in backtics:

```haskell
list1 `append` list2
```

Similarly, you can use operators in prefix form by taking them in parentheses:

```haskell
(+) x y
```

That's why you can eta-reduce here:

```haskell
append a b = a ++ b
append a b = (++) a b  -- it's the same as above
append a = (++) a      -- eta-reduce one argument
append = (++)          -- eta-reduce second argument